blocks/carousel/carousel.js

```diff
@@ -0,0 +1,452 @@
+    if (isDesktop) {
+      if (activeDotIndex === dots.length - 1) {
+        clearInterval(intervalId); // Stop if it's desktop and reaches the last dot
+        return;
+      }
+      activeDotIndex = (activeDotIndex + 1) % dots.length; // Move to the next dot
+    } else {
+      if (activeDotIndex === 0) {
+        movingForward = true; // Switch to moving forward
+      } else if (activeDotIndex === dots.length - 1) {
+        movingForward = false; // Switch to moving in reverse
+      }
+      activeDotIndex = movingForward ? (activeDotIndex + 1) % dots.length : activeDotIndex - 1;
+      if (activeDotIndex < 0) {
+        activeDotIndex = dots.length - 1;
+      }
+    }
+    const activeDot = dots[activeDotIndex];
+    updateCarouselView(activeDot);
+  }, 2000);
+}
+
+function setCarouselView(type, carouselSlider) {
+  const carouselTrack = carouselSlider.querySelector('.carousel-track');
+  const cards = Array.from(carouselTrack.children);
+  const visibleCards = allowedCardsCount();
+  const numberOfDots = cards.length - visibleCards + 1;
+  if (numberOfDots > 1) {
+    const dotsContainer = document.createElement('div');
+    dotsContainer.className = 'dots-container border-box';
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < numberOfDots; i++) {
+      const dot = document.createElement('button');
+      dot.className = 'dot border-box';
+      dot.dataset.index = i;
+      dotsContainer.appendChild(dot);
+      dot.addEventListener('click', function updateCarouselOnClick() {
+        updateCarouselView(this);
+      });
+    }
+
+    carouselSlider.appendChild(dotsContainer);
+    updateCarouselView(dotsContainer.firstChild);
+    startUpdateCarousel(carouselSlider);
+  }
+}
+
+function createDropdown(dropdownValue) {
+  const menuItems = dropdownValue.split(', ');
+  const dropdownText = menuItems[0];
+
+  const dropdownSelectDiv = document.createElement('div');
+  dropdownSelectDiv.className = 'dropdown-select border-box';
+
+  const button = document.createElement('button');
+  button.className = 'dropdown-toggle border-box';
+  button.innerHTML = `<span class="dropdown-text">${dropdownText}</span><span class="icon-down-arrow icon"></span>`;
+
+  const dropdownMenuContainer = document.createElement('div');
+  dropdownMenuContainer.className = 'dropdown-menu-container border-box';
+
+  const ul = document.createElement('ul');
+  ul.className = 'dropdown-menu border-box';
+
+  menuItems.forEach((itemText) => {
+    const li = document.createElement('li');
+    const a = document.createElement('a');
+    const span = document.createElement('span');
+    span.textContent = itemText;
+    a.appendChild(span);
+    li.appendChild(a);
+    ul.appendChild(li);
+  });
+
+  dropdownMenuContainer.appendChild(ul);
+  dropdownSelectDiv.appendChild(button);
+  dropdownSelectDiv.appendChild(dropdownMenuContainer);
+  button.addEventListener('click', () => {
+    if (dropdownMenuContainer.style.display === 'block') {
```

Comment:
Rather than applying `style` directly to elements, I'd suggest doing it via CSS classes. Also you can simplify this code a bit using toggle like `div.classList.toggle("visible");`
[Like here](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#examples)

Reply:
updated code with using class